developer/src/kmc-ldml/src/compiler/tran.ts

```diff
@@ -141,13 +144,21 @@ export abstract class TransformCompiler<T extends TransformCompilerType, TranBas
     // don't unescape literals here, because we are going to pass them through into the regex
     cookedFrom = util.unescapeStringToRegex(cookedFrom);
 
-    this.checkRanges(cookedFrom); // check before normalizing
+    // cook the ranges also
+    cookedFrom = this.checkRanges(cookedFrom); // check before normalizing
 
     if (!sections?.meta?.normalizationDisabled) {
       // nfd here.
       cookedFrom = MarkerParser.nfd_markers(cookedFrom, true);
     }
 
+    try {
+      new RegExp(cookedFrom, 'ug');
+    } catch (e) {
+      this.callbacks.reportMessage(CompilerMessages.Error_UnparseableTransformFrom({ from: transform.from }));
```

Comment:
`new RegExp` can return useful information about the error; do we want to extract that?

```suggestion
      this.callbacks.reportMessage(CompilerMessages.Error_UnparseableTransformFrom({ from: transform.from, e }));
```

![image](https://github.com/keymanapp/keyman/assets/4498365/8cdddbc6-6385-4afe-b597-21ff00a53659)

The exception messages returned are somewhat inconsistent -- some print the original regex, others don't.

Reply:
that's a good idea. Might be challenging on the unit test side, though, if the wording changes or is localized.

Reply:
unit tests probably shouldn't be checking the text of the error message, just the error code.